api/api.go

```diff
@@ -325,22 +339,41 @@ func (w *ApiWrapper) Verify(ctx echo.Context) error {
 	return ctx.JSON(http.StatusOK, verifyResponse)
 }
 
+// PublicKey returns a public key for the given urn. The urn represents a legal entity. The api returns the public key either in PEM or JWK format.
+// It uses the accept header to determine this. Default is PEM (text/plain), only when application/json is requested will it return JWK.
 func (w *ApiWrapper) PublicKey(ctx echo.Context, urn string) error {
 	if match, err := regexp.MatchString(`^\S+$`, urn); !match || err != nil {
 		return echo.NewHTTPError(http.StatusBadRequest, "incorrect organization urn in request")
 	}
-	pubKey, err := w.C.PublicKey(types.LegalEntity{URI: urn})
 
+	le := types.LegalEntity{URI: urn}
+	acceptHeader := ctx.Request().Header.Get("Accept")
+
+	// starts with so we can ignore any +
+	if strings.Index(acceptHeader, "application/json") == 0 {
+		jwk, err := w.C.PublicKeyInJWK(le)
+		if err != nil {
+			if strings.Contains(err.Error(), "could not open private key") {
+				return ctx.NoContent(404)
+			}
+			logrus.Error(err.Error())
+			return err
```

Comment:
The error is returned to echo where it will do its magic to convert it into a 500 and some logging. The logging can be removed, but then we first have to configure the echo logger properly.